### PR TITLE
shot at bringing draft spec up to date with adrs

### PR DESCRIPTION
**Pull request by nsheff**
Far from final, but this at least brings a draft specification up-to-date with our current decisions. Take a look if you can but the goal here is to provide a general, one-stop description of seqcol that could be useful to have during Connect.

I do still need to get the 'inherent' stuff in there, though.

TODO:
- [ ] Bring in some use cases (See: https://github.com/ga4gh/seqcol-spec/issues/11)
- [ ] Finalize the actual "minimal schema" we will propose.

### CODE REVIEW

File: docs/specification.md [hunk 1]
```diff
@@ -1,150 +1,217 @@
...
+
+The goal of the Seqcol project is **to standardize unique identifiers for collections of sequences**. It can be used to identify genomes, transcriptomes, or proteomes -- anything that can be represented as a collection of sequences. Seqcol uses a hash algorithm to generate a digest of the underlying collection of sequences. These unique identifiers are defined by an algorithm, rather than an accession authority, and are thus de-centralized and therefore usable for many purposes, including private or new sequence collections, cases without connection to a central database, or validation of sequence collection content and provenance.
```
Comment (tcezard) [inline docs/specification.md:23]: I'm not sure I would emphasise uniqueness here: seqcol identifiers are not primarily unique. For me the main goal is to create identifiers derived from the collection itself.

File: docs/specification.md [hunk 2]
```diff
@@ -1,150 +1,217 @@
...
+  "names": [
+    "chrX",
+    "chr1",
+    "chr2"
+  ],
+  "sequences": [
+    "5f63cfaa3ef61f88c9635fb9d18ec945",
+    "31fc6ca291a32fb9df82b85e5f077e31",
+    "92c6a56c9e9459d8a42b96f7884710bc"
+  ],
```
Comment (tcezard) [inline docs/specification.md:117]:
```suggestion
  ]
}
```

File: docs/specification.md [hunk 3]
```diff
@@ -1,150 +1,217 @@
...
+
+The object is a series of arrays with matching length (`3`), with the corresponding entries collated such that the first element of each array corresponds to the first element of each other array.
+
+
+#### Seqcol algorithm:
+
+Once the content of the sequence collection has been organized in the *seqcol object representation*, we can then run the digest algorithm to compute the unique identifier for this sequence collection
+
+1. **Step 1**. Put all attributes of the sequence collection (*e.g.* names, lengths, sequence digests) in seqcol object representation form, as defined above.
+2. **Step 2**. Apply [RFC-8785 JSON Canonicalization Scheme](https://www.rfc-editor.org/rfc/rfc8785) (JCS) to canonicalize the JSON of the seqcol object.
```
Comment (tcezard) [inline docs/specification.md:128]:
```suggestion
2. **Step 2**. Apply [RFC-8785 JSON Canonicalization Scheme](https://www.rfc-editor.org/rfc/rfc8785) (JCS) to canonicalize the value associated with each attributes individually.
```

File: docs/specification.md [hunk 4]
```diff
@@ -1,150 +1,217 @@
...
+3. **Step 3**. Digest the canonical representation of each attribute using the seqcol digest algorithm (defined below). This should convert the value of each attribute in the seqcol into a digest string.
```
Comment (tcezard) [inline docs/specification.md:129]:
```suggestion
3. **Step 3**. Digest the canonical representation of each attribute's value using the seqcol digest algorithm (defined below). This should convert the value of each attribute in the seqcol into a digest string.
```

**Comment by nsheff** [comment]
I will merge this to dev to make it easier to see and review this.

**Pull request merged**